### PR TITLE
Inventory

### PR DESCRIPTION
**Pull request by mpa-mxiang**
For this PR:

- Inventory&Inventory data model
- The above related database (cannot migrate successfully, **needs help**)
- Shopping cart controller and its route

### CODE REVIEW

File: db/migrate/20230920011502_create_inventory_foods.rb
Comment (emarkees) [inline db/migrate/20230920011502_create_inventory_foods.rb:1]: In line #5 the reference should be to inventory and not inventoryId

Kindly fix

File: db/migrate/20230920010531_create_inventories.rb
Comment (emarkees) [inline db/migrate/20230920010531_create_inventories.rb:1]: #line 5 also need to be fixed to just user and not userId

**Review by emarkees — CHANGES_REQUESTED** [review]
Hi 👍  @mpa-mxiang 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230920011502_create_inventory_foods.rb
Comment (mpa-mxiang) [reply db/migrate/20230920011502_create_inventory_foods.rb:1]: Fixed.

File: db/migrate/20230920010531_create_inventories.rb
Comment (mpa-mxiang) [reply db/migrate/20230920010531_create_inventories.rb:1]: Fixed

**Review by emarkees — APPROVED** [review]
Hi   @mpa-mxiang  
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linters errors
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**